BranchMetrics/cpp-branch-deep-linking-attribution
# SDK-625 -- generate a long link when the internet connection does not exist

## Reference
SDK-625 -- generate a long link when the internet connection does not exist

## Description
When the internet is off, link creation should fall back to "long link" creation.

Other SDKs use a timer to arbitrarily fall back after some number of seconds.  This implementation does two things
1. Put the Link request on the front of the queue
2. Don't retry if there is a failure.

## Testing Instructions
* Unit Tests Pass

* Test Case 1:  If there is no internet, should immediately (quickly) return a long link
* Test Case 2:  If "do not track" is turned on, should immediately return a long link

* Test Case 3:  With the example sample,
1. Turn Internet Off
2. Select Option "2" -- Send Session Open Event (or any option that puts something on the queue)
3. Note that this event will go into a retry loop
4. With the internet still off, select Option "c" -- Create a Branch Link

> Expected:  This should, after a few seconds return a long link.
> Note:  You may see a "success" log message when turning the internet back on -- but the callback will not be called the second time.

## Risk Assessment [`HIGH`]
Lots of base logic changes here


## Reviewer Checklist (To be checked off by the reviewer only)

- Correctness & Style
